modules/components/src/AdvancedFacetView/index.js

```diff
@@ -130,7 +131,7 @@ export default class AdvancedFacetView extends React.Component {
           aggregations,
         }).displayTreeDataWithValue
       : displayTreeData;
-
+      const filteredAggregations = matchingAggregations(aggregations, searchTerm);
```

Comment:
It's a patch, it doesn't solve the root problem, but it's the safest and simplest solution I could think of. I've been told that this whole part could be refactored so I didn't want to invest too much in it.